versioning/src/main/scala/org/typelevel/sbt/TypelevelVersioningPlugin.scala

```diff
@@ -82,17 +82,25 @@ object TypelevelVersioningPlugin extends AutoPlugin {
               None
           }
 
+        // version here is the prefix used further to build a final version number
         var version = latestInSeries.fold(tlBaseVersion.value)(_.toString)
 
-        // Looks for the distance to latest release in this series
-        latestInSeries.foreach { latestInSeries =>
-          Try(s"git describe --tags --match v$latestInSeries".!!.trim)
-            .collect { case Description(distance) => distance }
-            .foreach { distance => version += s"-$distance" }
+        V(version) match {
+          case None =>
+            sys.error(s"verion $version not formatted properly.")
+          case Some(value) =>
+            if (!(value.isSameSeries(baseV) || value >= baseV))
+              sys.error(s"Your latest tag $version cannot be less than tlBaseVersion $baseV")
+            else
+              // Looks for the distance to latest release in this series
+              latestInSeries.foreach { latestInSeries =>
+                Try(s"git describe --tags --match v$latestInSeries".!!.trim)
+                  .collect { case Description(distance) => distance }
+                  .foreach { distance => version += s"-$distance" }
+              }
+            git.gitHeadCommit.value.foreach { sha => version += s"-${sha.take(7)}" }
+            version
```

Comment:
Since this code is imperative anyway, do you mind if we leave these steps outside/after the `match` statement?

Reply:
we could however the if would in-exhuastive is that ok ?

Reply:
Yes, when the `if` doesn't match it just means the code should continue :) essentially we are writing

```scala
// do some stuff
if (isInvalid(stuff)) throw new Exception
// do more stuff
if (isInvalid(moreStuff)) throw new Exception
// return final answer
```